cpp/src/io/orc/writer_impl.cu

```diff
@@ -988,17 +991,19 @@ encoded_data encode_columns(orc_table_view const& orc_table,
   }
   chunk_streams.host_to_device(stream);
 
-  if (orc_table.num_string_columns() != 0) {
-    auto d_stripe_dict = orc_table.string_column(0).device_stripe_dict();
-    gpu::EncodeStripeDictionaries(d_stripe_dict,
-                                  chunks,
-                                  orc_table.num_string_columns(),
-                                  segmentation.num_stripes(),
-                                  chunk_streams,
-                                  stream);
-  }
+  if (orc_table.num_rows() > 0) {
+    if (orc_table.num_string_columns() != 0) {
```

Comment:
```suggestion
  if (orc_table.num_rows() > 0 and orc_table.num_string_columns() != 0) {
```

Reply:
I don't know enough to say whether this suggestion is correct or not, but it is a behavior change. With this change `gpu::EncodeStripeDictionaries` won't get called when there are nonzero rows without any string columns. Is that desirable?

Reply:
@vyasr did you mean `EncodeOrcColumnData`? If so, yes. We can't merge these conditions because the conditional branches have different scope, and we need to call `EncodeOrcColumnData` regardless of the number of string columns.

Reply:
Yeah sorry copy-pasted the wrong line, but that is what I meant. Merging these conditions removes that check and it seemed off to me.